geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java

```diff
@@ -693,15 +693,14 @@ public boolean put(Object object) throws InterruptedException, CacheException {
 
     boolean isDREvent = isDREvent(sender.getCache(), value);
 
-    Region region = value.getRegion();
-    String regionPath = null;
-    if (isDREvent) {
-      regionPath = region.getFullPath();
-    } else {
-      regionPath = ColocationHelper.getLeaderRegion((PartitionedRegion) region).getFullPath();
+    String regionPath = value.getRegionPath();
```

Comment:
Reading through ticket comments; it sounded like getRegion() can block when region is under GII or recovery...Is that expected, if not then the issue may be in how getRegion() is working...As I understand, application can operate on region even when the region GII is in progress.